// @marscode/baas-sdk/adapter/http/path.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.apiKeyToConfig = exports.baasApiKeys = void 0;
const config_1 = require("./config");
exports.baasApiKeys = {
    getToken: "baasServer:getToken",
    preSign: "baasServer:preSign",
    uploadFile: "baasServer:uploadFile",
    downloadFile: "baasServer:downloadFile",
    deleteFile: "baasServer:deleteFile",
    listFile: "baasServer:listFile",
    getFileByID: "baasServer:getFileByID",
    executeCmd: "baasServer:executeCmd",
};
exports.apiKeyToConfig = {
    [exports.baasApiKeys.getToken]: new config_1.PostHttpConfig(`/v1/baas/control/get_token`, exports.baasApiKeys.getToken),
    [exports.baasApiKeys.preSign]: new config_1.GetHttpConfig(`/v1/baas/data/file/api/resource/:ResourceID/put/presign`, exports.baasApiKeys.preSign),
    [exports.baasApiKeys.uploadFile]: new config_1.PutHttpConfig(``, exports.baasApiKeys.uploadFile),
    [exports.baasApiKeys.downloadFile]: new config_1.GetHttpConfig(``, exports.baasApiKeys.downloadFile),
    [exports.baasApiKeys.deleteFile]: new config_1.DeleteHttpConfig(`/v1/baas/data/file/api/resource/:ResourceID`, exports.baasApiKeys.deleteFile),
    [exports.baasApiKeys.listFile]: new config_1.PostHttpConfig(`/v1/baas/data/file/api/resource/:ResourceID/list`, exports.baasApiKeys.listFile),
    [exports.baasApiKeys.getFileByID]: new config_1.GetHttpConfig(`/v1/baas/data/file/api/resource/:ResourceID/file/:FileID`, exports.baasApiKeys.getFileByID),
    [exports.baasApiKeys.executeCmd]: new config_1.PostHttpConfig(`/v1/baas/data/redis/cmd/:ResourceID/:Cmd`, exports.baasApiKeys.executeCmd),
};
